aiida_common_workflows/workflows/relax/generator.py

```diff
@@ -32,7 +32,6 @@ def define(cls, spec):
         )
         spec.input(
             'protocol',
-            valid_type=ChoiceType(('fast', 'moderate', 'precise')),
```

Comment:
Good to discuss whether this should be limited, but this explicit definition did have the advantage that it made it discoverable exactly which protocols are available. If we remove this, we should reintroduce it in some other way.

Reply:
This is an interesting point. The valid protocols can be always extended (actually changed) inside the code implementations of the `CommonRelaxInputGenerator`, am I right @sphuber? This is compatible also with the concept of creating code-agnostic workchains only if we remove the choice check in the inputs of the workchain. Foor instance this is what we do in the EOS wc: the valid types disappear (https://github.com/aiidateam/aiida-common-workflows/blob/69468a77d30af4a0d7f03bfbe27e5ea4a605dbca/aiida_common_workflows/workflows/eos.py#L96). So in principle I do not see a drawback in maintaining this line.

Reply:
good point @bosonie. I should be able to just override it in `CastepCommonRelaxInputGenerator`. Will drop the changes to the base class.